fix TestCleanupExternalView.test

LOG 2901
>2020-10-10T05:38:01.4787071Z [ERROR] test(org.apache.helix.integration.TestCleanupExternalView)  Time elapsed: 1.462 s  <<< FAILURE!
2020-10-10T05:38:01.4894203Z java.lang.AssertionError: external-view for TestDB0 should be removed, but was: ZnRecord=TestDB0, {BUCKET_SIZE=0, IDEAL_STATE_MODE=AUTO, NUM_PARTITIONS=2, REBALANCE_MODE=SEMI_AUTO, REBALANCE_STRATEGY=DEFAULT, REPLICAS=2, STATE_MODEL_DEF_REF=MasterSlave, STATE_MODEL_FACTORY_NAME=DEFAULT}{TestDB0_0={localhost_12918=SLAVE, localhost_12919=MASTER}, TestDB0_1={localhost_12918=MASTER, localhost_12919=SLAVE}}{}, Stat=Stat {_version=6, _creationTime=1602305095151, _modifiedTime=1602305095369, _ephemeralOwner=0} expected:<true> but was:<false>
2020-10-10T05:38:01.4970061Z 	at org.apache.helix.integration.TestCleanupExternalView.test(TestCleanupExternalView.java:122)
2020-10-10T05:38:01.4972687Z 
2020-10-10T05:38:01.8976500Z [ERROR] Failures: 
2020-10-10T05:38:01.8991408Z [ERROR]   TestCleanupExternalView.test:122 external-view for TestDB0 should be removed, but was: ZnRecord=TestDB0, {BUCKET_SIZE=0, IDEAL_STATE_MODE=AUTO, NUM_PARTITIONS=2, REBALANCE_MODE=SEMI_AUTO, REBALANCE_STRATEGY=DEFAULT, REPLICAS=2, STATE_MODEL_DEF_REF=MasterSlave, STATE_MODEL_FACTORY_NAME=DEFAULT}{TestDB0_0={localhost_12918=SLAVE, localhost_12919=MASTER}, TestDB0_1={localhost_12918=MASTER, localhost_12919=SLAVE}}{}, Stat=Stat {_version=6, _creationTime=1602305095151, _modifiedTime=1602305095369, _ephemeralOwner=0} expected:<true> but was:<false>
2020-10-10T05:38:01.9001717Z [ERROR] Tests run: 1211, Failures: 1, Errors: 0, Skipped: 0
2020-10-10T05:38:01.9107617Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project helix-core: There are test failures.
2020-10-10T05:38:01.9110958Z [ERROR]

--- a/helix-core/src/test/java/org/apache/helix/integration/TestCleanupExternalView.java
+++ b/helix-core/src/test/java/org/apache/helix/integration/TestCleanupExternalView.java
@@ -109,18 +109,14 @@ public class TestCleanupExternalView extends ZkUnitTestBase {
     // System.out.println("re-enabling controller");
     admin.enableCluster(clusterName, true);
 
-    ExternalView externalView = null;
-    for (int i = 0; i < 10; i++) {
-      Thread.sleep(100);
-      externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
-      // System.out.println("externalView: " + externalView);
-      if (externalView == null) {
-        break;
-      }
-    }
 
-    Assert.assertNull(externalView, "external-view for TestDB0 should be removed, but was: "
-        + externalView);
+    result = TestHelper.verify(() -> {
+      ExternalView externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
+      return externalView == null;
+    }, TestHelper.WAIT_DURATION);
+
+    ExternalView ev = accessor.getProperty(keyBuilder.externalView("TestDB0"));
+    Assert.assertTrue(result, "external-view for TestDB0 should be removed, but was: " + ev);
 
     // clean up
     controller.syncStop();

--- a/helix-core/src/test/java/org/apache/helix/integration/TestCleanupExternalView.java
+++ b/helix-core/src/test/java/org/apache/helix/integration/TestCleanupExternalView.java
@@ -42,6 +42,7 @@ import org.testng.annotations.Test;
  * orphan external-view
  */
 public class TestCleanupExternalView extends ZkUnitTestBase {
+  private ExternalView _externalView = null;
   @Test
   public void test() throws Exception {
     // Logger.getRootLogger().setLevel(Level.INFO);
@@ -109,14 +110,12 @@ public class TestCleanupExternalView extends ZkUnitTestBase {
     // System.out.println("re-enabling controller");
     admin.enableCluster(clusterName, true);
 
-
     result = TestHelper.verify(() -> {
-      ExternalView externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
-      return externalView == null;
+      _externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
+      return _externalView == null;
     }, TestHelper.WAIT_DURATION);
 
-    ExternalView ev = accessor.getProperty(keyBuilder.externalView("TestDB0"));
-    Assert.assertTrue(result, "external-view for TestDB0 should be removed, but was: " + ev);
+    Assert.assertTrue(result, "external-view for TestDB0 should be removed, but was: " + _externalView);
 
     // clean up
     controller.syncStop();

